### ISSUE TITLE
Can't upload a Package Zipfile to the Playground

### ISSUE THREAD

**Issue by BryceStevenWilley** [jhpyle/docassemble#385]
I've seen quite a few examples of this, and can reproduce on 1.2.41.

```
Bad Zipfile when trying to upload a zip: zip file with `zipfile` in python is okay locally, but error is:
"""BadZipFile: File is not a zip file...
...
  File "/usr/share/docassemble/local3.8/lib/python3.8/site-packages/docassemble/webapp/server.py", line 17785, in playground_package...
    with zipfile.ZipFile(zippath, mode='r') as zf:
  File "user/lib/python3.8/zipfile.py", line 1269, in __init__
    self._RealGetContents()
  File "/usr/lib/python3.8/zipfil.py", line 1334, in _RealGetContents
    raise BadZipFile("File is not a zip file")
zipfile.BadZipFile: File is not a zip file
```

Seems to be related to https://github.com/jhpyle/docassemble/commit/4ba0510d507a332dcc8efc4c7390d6a60f26acb5#diff-4c8bf136c393acf44c18098d2b935b3a005636e57574530acffdf552432360faR17766, but I'm not sure. I have a feeling that things aren't properly being written to the TmpFile object, but I want to check to see that's the case.

**Comment by nonprofittechy** [comment]
I also ran into this recently--downloaded a package from playground, was not able to upload it.

**Comment by BryceStevenWilley** [comment]
I can confirm that it's an issue with the contents being written to the file, but not to the file object. Testing a fix at the moment, PR coming soon.

**Comment by jhpyle** [comment]
Hopefully this is fixed in 1.2.46.

**Comment by BryceStevenWilley** [comment]
You beat me to it! But yep, I can confirm that 1.2.46 fixes that (and the issue with downloading Zips from the playground that I had yet to look into!)
